### ISSUE TITLE
BuildInfoSource incompatible with functions/API Gateway

### ISSUE THREAD

**Issue by cherrydev** [micronaut-projects/micronaut-core#1747]
### Task List

- [ ] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. Configure a RequestStreamHandler Lambda with `MicronautLambdaContainerHandler.getAwsProxyHandler()` in a serverless or CloudFormation environment. An example is [here](https://github.com/micronaut-projects/micronaut-aws/tree/master/examples/api-gateway-example).
2. include `io.micronaut:micronaut-management`
3. add to `application.yml`:
```
endpoints:
  info:
    enabled: true
    sensitive: false
```
4. Navigate to the `/info` endpoint, relative to the API Gateway endpoint

### Expected Behaviour

Expected an info endpoint result.

### Actual Behaviour

An exception is logged in CloudWatch and the endpoint returns a 502 error.
```
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [refreshScope] of class: io.micronaut.runtime.context.scope.refresh.RefreshInterceptor
Message: No bean of type [io.micronaut.runtime.context.scope.refresh.RefreshScope] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new InfoEndpoint(InfoAggregator infoAggregator,[InfoSource[] infoSources]) --> new $BuildInfoSourceDefinition$Intercepted(ResourceResolver resourceResolver,String buildPropertiesPath,BeanContext beanContext,[Interceptor[] interceptors]) --> new RefreshInterceptor([RefreshScope refreshScope])
```

All AWS functions (with or without API Gateway) run with the `Environment.FUNCTION` environment. The built-in `BuildInfoSource` is a `@Refreshable` bean and `RefreshScope` is disabled when running in `Environment.FUNCTION`. Therefore this could probably be easily fixed by creating an abstract base class of the `BuildInfoSource` bean and two concrete versions, one that's `@Refreshable` and `@Requires(notEnv = Environment.FUNCTION)` and the other `@Requires(notEnv = Environment.FUNCTION)` and NOT `@Refreshable`.

### Example Application

https://github.com/micronaut-projects/micronaut-aws/tree/master/examples/api-gateway-example

**Comment by graemerocher** [comment]
Yeah the management module doesn't currently support serverless functions. Not sure why you would want it as functions are short lived and the kind of information they expose (metrics, environment etc.) goes away
